### PR TITLE
Surface VAT options for the duty calculator

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

https://transformuk.atlassian.net/browse/HOTT-592

### What?

I have added/removed/altered:

- [x] Adds ApplicableVatOptionsService and specs
- [x] Update business logic for determining VAT measure types
- [x] Surface vat options for the duty calculator

### Why?

I am doing this because:

- These are needed for calculating VAT measures

### CODE REVIEW

File: app/models/measure_type.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ class MeasureType < Sequel::Model
   IMPORT_MOVEMENT_CODES = [0, 2].freeze
   EXPORT_MOVEMENT_CODES = [1, 2].freeze
   THIRD_COUNTRY = %w[103 105].freeze # 105 measure types are for end use Third Country duties. 103 are for everything else
-  VAT_TYPES = %w[VTA VTE VTS VTZ 305].freeze
+  VAT_TYPES = %w[305].freeze
```
Comment (willfish) [inline app/models/measure_type.rb:9]: The measure_type_id would never have been VTA, VTE, VTS or VTZ from my observations of the data.

Comment (tavy88) [reply app/models/measure_type.rb:9]: Ok, check with Matt just in case.

**Review by tavy88 — APPROVED** [review]

**Comment by willfish** [comment]
```
echo 1502000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1502000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 8908000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/8908000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 6307901000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/6307901000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 6500000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/6500000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 4901000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/4901000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1207999600; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1207999600 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 4903000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/4903000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 2006003800; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/2006003800 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 2006009100; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/2006009100 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 2006009900; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/2006009900 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1504000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1504000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 4906000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/4906000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 3926909790; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/3926909790 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 3006600000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/3006600000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 0206220000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/0206220000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 2309000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/2309000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1506000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1506000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 2924297099; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/2924297099 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 0301930000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/0301930000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 0106410000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/0106410000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 0106490000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/0106490000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1207100000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1207100000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 0106900000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/0106900000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 0206300000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/0206300000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1509000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1509000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1510000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1510000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 8716109800; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/8716109800 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 2934999090; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/2934999090 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1213000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1213000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 2202000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/2202000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1207300000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1207300000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1702500000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1702500000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 0206410000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/0206410000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 0206490000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/0206490000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 0106330000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/0106330000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 3926200000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/3926200000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1702905000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1702905000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1207600000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1207600000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 2301000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/2301000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1207700000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1207700000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 8902000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/8902000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 3002120000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/3002120000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 8904000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/8904000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 0602400000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/0602400000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1702908000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1702908000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1702909500; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1702909500 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1209999900; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1209999900 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 0106190000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/0106190000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 0106130000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/0106130000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 0602903000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/0602903000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 1501000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/1501000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
echo 9900000000; curl -s https://dev.trade-tariff.service.gov.uk/api/v2/commodities/9900000000 | jq '.data.meta.duty_calculator.applicable_vat_options' 2> /dev/null; echo
```

File: app/models/measure_type.rb (hunk 1 above)
Comment (willfish) [reply app/models/measure_type.rb:9]: XI:

```sql
SELECT COUNT(1) FROM measures WHERE measure_type_id IN ('VTA', 'VTE', 'VTS', 'VTZ');
 count
--------
 386433
(1 row)
```

UK
```sql
SELECT COUNT(1) FROM measures WHERE measure_type_id IN ('VTA', 'VTE', 'VTS', 'VTZ');
 count
-------
     0
(1 row)
```

**Pull request merged**